samcli/commands/check/calculations.py

```diff
@@ -0,0 +1,125 @@
+"""
+Bottle neck calculations are done here. Warning messages are also generated here
+"""
+import click
+import boto3
+import botocore
+
+
+from samcli.commands._utils.resources import AWS_LAMBDA_FUNCTION
+from samcli.commands.check.resources.Graph import Graph
+from .resources.Warning import CheckWarning
+
+
+class Calculations:
+    _graph: Graph
+
+    def __init__(self, graph: Graph):
+        """
+        Args:
+            graph (Graph): The graph object. This is where all of the data is stored
+        """
+        self._graph = graph
+
+    def _generate_warning_message(
+        self,
+        capacity_used: int,
+        resource_name: str,
+        concurrent_executions: int,
+        duration: int,
+        tps: int,
+        burst_concurrency: int,
+    ):
+        """Depending on how severe the bottle neck is, a different warning message will be generated
+        Args:
+            capacity_used (int): Percentage of capacity used
+            resource_name (str): Recource name that the warning is for
+            concurrent_executions (int): Concurrent executions based on users account
+            duration (int): Duration of lambda function
+            tps (int): TPS of lambda function based on its parent resource
+            burst_concurrency (int): Burst concurrency based on users account
+        """
+        warning = CheckWarning()
+
+        if capacity_used <= 70:
+            warning.message = (
+                "For the lambda function [%s], you will not be close to its soft limit of %i concurrent executions."
+                % (resource_name, concurrent_executions)
+            )
+            self._graph.green_warnings.append(warning)
+
+        elif capacity_used < 90:
+            warning.message = (
+                "For the lambda function [%s], the %ims duration and %iTPS arrival rate is using %i%% of the allowed "
+                + "concurrency on AWS Lambda. A limit increase should be considered:"
+                + "\nhttps://console.aws.amazon.com/servicequotas"
+            ) % (resource_name, duration, tps, round(capacity_used))
+            self._graph.yellow_warnings.append(warning)
+
+        elif capacity_used <= 100:
+            warning.message = (
+                "For the lambda function [%s], the %ims duration and %iTPS arrival rate is using %i%% of the allowed "
+                + "concurrency on AWS Lambda. It is very close to the limits of the lambda function. It is strongly "
```

Comment:
The `+` sign is not necessary here. Consecutive lines of strings will just concatenate into one when they are in the same `()`. For example,
```
s = (
    "hello, "
    "world!"
)
print(s) # 'hello, world1'
```